src/CourseModule/controllers/course.controller.ts

```diff
@@ -113,4 +115,19 @@ export class CourseController {
   public async delete(@Param('id') id: CourseDTO['id']): Promise<void> {
     await this.service.delete(id);
   }
+
+  @Get('author/:name')
+  @HttpCode(200)
+  @ApiOkResponse({ type: NewUserDTO })
+  @ApiImplicitQuery({ name: 'name', type: Number, required: true, description: 'Author name' })
+  @ApiOperation({ title: 'Find courses by author name', description: 'Find courses by author name' })
+  @ApiNotFoundResponse({ description: 'thrown if courses are not found' })
+  @ApiUnauthorizedResponse({ description: 'thrown if there is not an authorization token or if authorization token does not have ADMIN, STUDENT or EXTERNAL role' })
+  @NeedRole(RoleEnum.ADMIN, RoleEnum.STUDENT, RoleEnum.EXTERNAL)
+  @UseGuards(RoleGuard)
+  public async findUserCoursesById(
```

Comment:
```suggestion
  public async findUserCoursesByAuthorName(
```
O método não busca pelo id!

Reply:
findCoursesByAuthorName só

Reply:
:+1: